databricks_cli/stack/api.py

```diff
@@ -0,0 +1,340 @@
+        click.echo("Deploying job '%s' with settings: \n%s \n" % (resource_id, json.dumps(
+            job_settings, indent=2, separators=(',', ': '))), nl=False)
+
+        if physical_id:
+            job_id = physical_id['job_id']
+            self.update_job(job_settings, physical_id['job_id'])
+        else:
+            job_id = self.put_job(job_settings)
+        click.echo("Job deployed on Databricks with job_id %s" % job_id)
+        physical_id = {'job_id': job_id}
+        deploy_output = self.jobs_client.get_job(job_id)
+        return physical_id, deploy_output
+
+    def deploy_resource(self, resource, resource_deploy_info=None):  # overwrite to be added
+        """
+        Deploys a resource given a resource information extracted from the stack JSON configuration
+        template.
+
+        :param resource: A dict of the resource with fields of RESOURCE_ID, RESOURCE_SERVICE and
+        RESOURCE_PROPERTIES.
+        ex. {'id': 'example-resource', 'service': 'jobs', 'properties': {...}}
+        :param resource_deploy_info: A dict of the resource's deployment info from the last
+        deployment. Will be None if this is the first deployment.
+        ex. {'id': 'example-resource', 'service': 'jobs', 'physical_id': {...}}
+        :return: dict resource_deploy_info- A dictionary of deployment information of the
+        resource to be stored at deploy time. It includes the resource id of the resource along
+        with the physical id and deploy output of the resource.
+        ex. {'id': 'example-resource', 'service': 'jobs', 'physical_id': {'job_id': 123},
+        'timestamp': 123456789, 'deploy_output': {..}}
+        """
+        # try:
+        resource_id = resource[RESOURCE_ID]
+        resource_service = resource[RESOURCE_SERVICE]
+        resource_properties = resource[RESOURCE_PROPERTIES]
+        # except KeyError as e:
+        #     raise StackError("%s doesn't exist in resource config" % str(e))
+
+        # Get physical ID for resource from last deployment
+        # physical_id = self._get_previous_resource_deploy_map(resource_id, resource_service)
+
+        physical_id = resource_deploy_info[RESOURCE_PHYSICAL_ID] if resource_deploy_info else None
+        if resource_service == JOBS_SERVICE:
+            physical_id, deploy_output = self.deploy_job(resource_id, resource_properties,
+                                                         physical_id)
+        else:
+            raise StackError("Resource service '%s' not supported" % resource_service)
+
+        resource_deploy_info = {RESOURCE_ID: resource_id, RESOURCE_SERVICE: resource_service,
+                                RESOURCE_DEPLOY_TIMESTAMP: datetime.now(),
+                                RESOURCE_PHYSICAL_ID: physical_id,
+                                RESOURCE_DEPLOY_OUTPUT: deploy_output}
+        return resource_deploy_info
+
+    def validate_status(self, stack_status):
```

Comment:
Why do we need to validate the status generated on runtime? This seems like something that should be validated during unit tests?